server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2229,7 +2229,7 @@ private void updateTemplate(long templateId, String fileExtension, long zoneId,
         VMTemplateVO template = _tmpltDao.findById(templateId);
         templateStore.setDownloadState(Status.DOWNLOADED);
         templateStore.setState(ObjectInDataStoreStateMachine.State.Ready);
-        templateStore.setInstallPath("template/tmpl/1/" + templateId + "/" + template.getUuid() + "." + fileExtension);
+        templateStore.setInstallPath("template/tmpl/2/" + templateId + "/" + template.getUuid() + "." + fileExtension);
```

Comment:
in a method called 'updateTemplate(...)' a hardcoded path like this is very suspect. Did you name the method correctly? it seems we need to give a more descriptive/restrictive/deterring name.